LongestSubstring.java

```diff
@@ -0,0 +1,38 @@
+/*
```

Comment:
Hey, thanks for sharing your code. I ran this against a few test cases, and it past most of the them, however it seems to fail when trying to process strings with repeating characters. 
## Walkthrough test case with input "123145678"

- _i = 0  substring = "" s[0] = "1"_
`true && true -> substring = "1"`
- _i = 1, substring ="1", s[1] = "2"_
`true && true -> substring = "12"`
- _i = 2, substring "12", s[2] = "3"_
`true && true -> substring = "123"`
- _i = 3, substring "123", s[3] = "1"_
`false && true -> substring = "123"`
- _i = 4, substring "123", s[4] = "4"_
`true && true -> substring = "1234"`
- _i = 5, substring "1234", s[5] = "5"_
`true && false -> substring = "1234"`
- _i = 6, substring "1234", s[6] = "6"_
`true && false -> substring = "1234"`
- _i = 7, substring "1234", s[7] = "7"_
`true && false -> substring = "1234"`
- _i = 8, substring "1234", s[8] = "8"_
`true && false -> substring = "1234"`